gnomad/assessment/sanity_checks.py

```diff
@@ -190,3 +207,766 @@ def compare_row_counts(ht1: hl.Table, ht2: hl.Table) -> bool:
+
+    var_summary = hl.summarize_variants(t, show=False)
+    logger.info(
+        "Dataset has %d variants distributed across the following contigs: %s",
+        var_summary.n_variants,
+        var_summary.contigs,
+    )
+
+    for contig in var_summary.contigs:
+        if var_summary.contigs[contig] == 0:
+            logger.warning("%s has no variants called", var_summary.contigs)
+
+    if monoallelic_expr is not None:
+        if isinstance(t, hl.MatrixTable):
+            mono_sites = t.filter_rows(monoallelic_expr).count_rows()
+        else:
+            mono_sites = t.filter(monoallelic_expr).count()
+        logger.info("There are %d monoallelic sites in the dataset.", mono_sites)
+
+    return var_summary
+
+
+def raw_and_adj_sanity_checks(
+    t: Union[hl.MatrixTable, hl.Table],
+    subsets: List[str],
+    verbose: bool,
+    delimiter: str = "-",
+    metric_first_label: bool = True,
+) -> None:
+    """
+    Perform sanity checks on raw and adj data in input Table/MatrixTable.
+
+    Check that:
+        - Raw AC, AN, AF are not 0
+        - Adj AN is not 0 and AC and AF are not negative
+        - Raw values for AC, AN, nhomalt in each sample subset are greater than or equal to their corresponding adj values
+
+    Raw and adj call stat annotations must be in an info struct annotation on the Table/MatrixTable, e.g. t.info.AC-raw.
+
+    :param t: Input MatrixTable or Table to check.
+    :param subsets: List of sample subsets.
+    :param verbose: If True, show top values of annotations being checked, including checks that pass; if False, show only top values of annotations that fail checks.
+    :param delimiter: String to use as delimiter when making group label combinations.
+    :param metric_first_label: If True, metric precedes label group, e.g. AC-afr-male. If False, label group precedes metric, afr-male-AC.
+    :return: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    field_check_expr = {}
+    field_check_details = {}
+    for subfield in ["AC", "AF"]:
+        # Check raw AC, AF > 0
+        check_field = f"{subfield}{delimiter}raw"
+        field_check_expr, field_check_details = make_field_check_dicts(
+            field_check_expr=field_check_expr,
+            field_check_details=field_check_details,
+            check_description=f"{check_field} > 0",
+            cond_expr=t.info[check_field] <= 0,
+            display_fields=hl.struct(**{check_field: t.info[check_field]}),
+        )
+
+        check_field = f"{subfield}{delimiter}adj"
+        field_check_expr, field_check_details = make_field_check_dicts(
+            field_check_expr=field_check_expr,
+            field_check_details=field_check_details,
+            check_description=f"{check_field} >= 0",
+            cond_expr=t.info[check_field] < 0,
+            display_fields=hl.struct(
+                **{check_field: t.info[check_field], "filters": t.filters}
+            ),
+        )
```

Comment:
PART OF LARGE SUGGESTION
```suggestion

        field_check_expr[f"{check_field} > 0"] = {
            "expr": hl.agg.count_where(t.info[check_field] <= 0),
            "display_fields": hl.struct(**{check_field: t.info[check_field]}),
        }

        check_field = f"{subfield}{delimiter}adj"
        field_check_expr[f"{check_field} >= 0"] = {
            "expr": hl.agg.count_where(t.info[check_field] < 0),
            "display_fields": hl.struct(
                **{check_field: t.info[check_field], "filters": t.filters}
            ),
        }
```